beefy-gadget/src/keystore.rs

```diff
@@ -39,10 +41,18 @@ impl BeefyKeystore {
 	pub fn authority_id(&self, keys: &[Public]) -> Option<Public> {
 		let store = self.0.clone()?;
 
-		for key in keys {
-			if SyncCryptoStore::has_keys(&*store, &[(key.to_raw_vec(), KEY_TYPE)]) {
-				return Some(key.clone());
-			}
+		// we do check for multiple private keys as a key store sanity check.
+		let public: Vec<Public> = keys
+			.iter()
+			.filter(|k| SyncCryptoStore::has_keys(&*store, &[(k.to_raw_vec(), KEY_TYPE)]))
+			.cloned()
+			.collect();
+
+		if public.len() > 1 {
+			warn!(target: "beefy", "🥩 Multiple private keys found for: {:?}", public);
+			return Some(public[0].clone());
+		} else if !public.is_empty() {
+			return Some(public[0].clone());
```

Comment:
I have opted for this check instead of checking the signed commitment right after signing it.

Main reason is that I do see a potential issue with multiple private keys if one is an ephemeral (in-memory) and the other is a persistent private key. The in-memory private key will *always* be the first one returned by *has_keys()*. So it could happen that the sign the commitment with the in-memory private key which will be gone after a node re-start. 

The above check together with `verify_validator_set()` will hopefully allow us to identify those cases.

Reply:
It still doesn't cover a case where keystore thinks has the keys, but the keys are simply invalid. I've added a check when `trace` logging is enabled.